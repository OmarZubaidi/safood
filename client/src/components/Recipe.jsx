import React from 'react';
import { Card } from "react-bootstrap";

export default function Recipe ({ recipe }) {
  return (
    <>
      <Card key={recipe.title}>
        <Card.Img variant="top" src={recipe.image} />
        <Card.Body>
          <Card.Title>{recipe.title}</Card.Title>
          <Card.Text>
            {recipe.ingredients}
          </Card.Text>
        </Card.Body>
        <Card.Footer>
          <small className="text-muted">Prep-time: {recipe.readyInMinutes} minutes</small>
        </Card.Footer>
      </Card>
    </>
  );
}
