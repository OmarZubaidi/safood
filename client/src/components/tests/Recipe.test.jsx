// Package imports
import { render, screen } from '@testing-library/react';

// Local imports
import Recipe from '../Recipe';

describe('Recipe component', () => {
  const mockRecipe = {
    "title": "Fried Anchovies with Sage",
    "readyInMinutes": "45",
    "image": "https://spoonacular.com/recipeImages/1-556x370.jpg",
    "ingredients": "anchovies,baking powder,egg,flour,sage leaves,salt,seltzer water,vegetable oil",
  };

  test('should match the snapshot', () => {
    const { container } = render(<Recipe recipe={mockRecipe} />);
    expect(container).toMatchSnapshot();
  });

  test('should render the recipe', () => {
    render(<Recipe recipe={mockRecipe} />);
    screen.getByText(mockRecipe.title);
    screen.getByText(`Prep-time: ${mockRecipe.readyInMinutes} minutes`);
    screen.getByText(`${mockRecipe.ingredients.split(',')[1]},`);
  });
});