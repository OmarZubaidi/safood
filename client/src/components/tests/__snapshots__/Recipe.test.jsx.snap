// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      class="card-img"
      src="https://spoonacular.com/recipeImages/1-556x370.jpg"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Fried Anchovies with Sage
      </div>
      <p
        class="card-text"
        style="min-height: 50px;"
      >
        <span>
          Anchovies, 
        </span>
        <span>
          baking powder, 
        </span>
        <span>
          egg, 
        </span>
        <span>
          flour, 
        </span>
        <span>
          sage leaves, 
        </span>
        <span>
          salt, 
        </span>
        <span>
          seltzer water, 
        </span>
        <span>
          vegetable oil.
        </span>
      </p>
    </div>
    <div
      class="card-footer"
    >
      <small
        class="text-muted"
      >
        Prep-time: 
        45
         minutes
      </small>
    </div>
  </div>
</div>
`;
